eng/common/pipelines/templates/steps/update-docsms-metadata.yml

```diff
@@ -0,0 +1,106 @@
+# intended to be used as part of a release process
+parameters:
+  - name: Artifacts
+    type: object
+    default: []
+  - name: RepoId
+    type: string
+    default: $(Build.Repository.Name)
+  - name: WorkingDirectory
+    type: string
+    default: ''
+  - name: ScriptDirectory
+    type: string
+    default: eng/common/scripts
+  - name: TargetDocRepoName
+    type: string
+    default: ''
+  - name: TargetDocRepoOwner
+    type: string
+  - name: Language
+    type: string
+    default: ''
+  - name: DailyDocsBuild
+    type: boolean
+    default: false
+  - name: SparseCheckoutPaths
+    type: object
+    default:
+      - '**'
+
+steps:
+- template: /eng/common/pipelines/templates/steps/enable-long-path-support.yml
+
+- pwsh: |
+    Write-Host "###vso[task.setvariable variable=DocRepoLocation]${{ parameters.WorkingDirectory }}/doc"
+  displayName: Set $(DocRepoLocation)
+
+- template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
+  parameters:
+    SkipDefaultCheckout: true
+    Repositories:
+      - Name: ${{ parameters.TargetDocRepoOwner }}/${{ parameters.TargetDocRepoName }}
+        WorkingDirectory: $(DocRepoLocation)
+    Paths: ${{ parameters.SparseCheckoutPaths }}
+
+# If performing a daily docs build set the $(TargetBranchName) to a daily branch
+# name and attempt to checkout the daily docs branch. If the branch doesn't
+# exist, create it
+- ${{ if eq(parameters.DailyDocsBuild, 'true') }}:
+  - template: /eng/common/pipelines/templates/steps/set-daily-docs-branch-name.yml
+    parameters:
+      DailyBranchVariableName: TargetBranchName
+
+  - pwsh: |
+      $ErrorActionPreference = "Continue"
+      $RemoteName = "origin"
+      $BranchName = "$(TargetBranchName)"
+      # Fetch and checkout remote branch if it already exists otherwise create a new branch.
+      git ls-remote --exit-code --heads $RemoteName $BranchName
+      if ($LASTEXITCODE -eq 0) {
```

Comment:
Do you need to do this step for the default branch as well? Or is the assumption that it will already be updated and checked out?

Reply:
The assumption is that the default branch has already been checked out from `sparse-checkout.yml`.